File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let candidate = 0, vote = 0;
    for(var num of nums){
        if(vote === 0) candidate = num;
        vote += (candidate === num) ? 1 : -1;
    }
    return candidate;
};

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    // Boyer–Moore majority vote algorithm
    int majorityElement(vector<int>& nums) {
        int candidate = 0, vote = 0;
        for (int num : nums) {
            if (vote == 0) {
                candidate = num;
            }
            vote += (num == candidate) ? 1 : -1;
        }
        return candidate;
    }
};